### PR TITLE
notmuch: use query as name for vfolder-from-query

### LINKED ISSUE #2033: Prettier status information for virtual folders created from notmuch queries
## What would you like to be able to do?

Show prettier status information for virtual folders created from notmuch queries (\<vfolder-from-query\>, a command which I use a lot)  instead of the encoded URI.

EDIT: Also possibly a prettier and more concise folder name in the index.

## What needs to change in NeoMutt?

The %f sequence could simply show the notmuch query.

EDIT: The %b sequence in the index could show something like "<notmuch>" instead of the URI.

## Have you tried any alternative methods?

From what I could gather, pretty names are currently only possible with named virtual mailboxes, which only accept the cumbersome URI format. 

## Additional Info

Annotation 2019-12-06 000003_blurred

[nm_pretty](https://github.com/lbrayner/neomutt/tree/nm_pretty) is an implementation of this idea. I patched *status.c*, *notmuch/mutt_notmuch.c*, and *notmuch/mutt_notmuch.h*. The  arguments of the notmuch URI (in `m->realpath`) are extracted and decoded with `nm_decode_uri`, a new function in *notmuch/mutt_notmuch.c*.

EDIT: I also patched *hdrline.c* to have the string "\<notmuch\>" appear for the %b sequence.

This is what it looks like upstream:

Annotation 2019-12-06 001514_blurred

### PR DESCRIPTION
**Pull request by Austin-Ray**
To provide a consistent experience for those who want human-readable
names, set the 'vfolder-from-query' mailbox's name to the user's query.

The previous behavior for 'vfolder-from-query' delegated mailbox
creation to lower-level backend-agnostic code, which lacks the ability
to set a name. As such, 'vfolder-from-query' mailboxes displayed the
encoded notmuch URI when the user wanted a human readable name.

Since we know a Mailbox will be always created from a valid
'vfolder-from-query' execution, we can create it at a high-level and
pass it to the lower-level components, which obviates the need for the
lower-level component to create it.

Closes #2033

### CODE REVIEW

**Comment by Austin-Ray** [comment]
Amended commit fixed error in commit message. Accidentally said "Fixes" instead of "Closes".

File: index.c [hunk 1]
```diff
@@ -2285,18 +2285,28 @@ int mutt_index_menu(struct MuttWindow *dlg)
...
           break;
         }
-        if (!nm_uri_from_query(NULL, buf, sizeof(buf)))
-          mutt_message(_("Failed to create query, aborting"));
+
+        char *query_unencoded = mutt_str_strdup(buf);
+
+        if (nm_uri_from_query(NULL, buf, sizeof(buf)))
+        {
+          struct Mailbox *temp_mail = mx_path_resolve(buf);
```
Comment (flatcap) [inline index.c]: (still unwritten) rule: Mailboxes are called `m` or `m_*`

File: index.c [hunk 2]
```diff
@@ -2285,18 +2285,28 @@ int mutt_index_menu(struct MuttWindow *dlg)
...
+          temp_mail->name = mutt_str_strdup(query_unencoded);
```
Comment (flatcap) [inline index.c]: We could make this a little more efficient:

```suggestion
          temp_mail->name = query_unencoded;
          query_unencoded = NULL;
```

**Comment by flatcap** [comment]
> Accidentally said "Fixes" instead of "Closes".

Both are fine.  [GitHub's fairly flexible](https://help.github.com/en/github/managing-your-work-on-github/closing-issues-using-keywords).

**Comment by Austin-Ray** [comment]
Addressed code review comments, and added comments for clarity. PR should be good to squash and merge. 

> Both are fine. [GitHub's fairly flexible](https://help.github.com/en/github/managing-your-work-on-github/closing-issues-using-keywords).

I'm just pedantic and want to properly reflect that this commit isn't a bug fix.

File: index.c [hunk 3]
```diff
@@ -2285,18 +2285,31 @@ int mutt_index_menu(struct MuttWindow *dlg)
...
+          struct Mailbox *m_new_vfolder = mx_path_resolve(buf);
+          m_new_vfolder->name = query_unencoded;
+          query_unencoded = NULL;
+
+          main_change_folder(menu, op, m_new_vfolder, buf, sizeof(buf), &oldcount, &index_hint, NULL);
+        }
         else
-          main_change_folder(menu, op, NULL, buf, sizeof(buf), &oldcount, &index_hint, NULL);
-        break;
+          mutt_message(_("Failed to create query, aborting"));
```
Comment (flatcap) [inline index.c]: Error path leaks `query_unencoded`

File: index.c [hunk 4]
```diff
@@ -2285,18 +2285,31 @@ int mutt_index_menu(struct MuttWindow *dlg)
...
         }
-        if (!nm_uri_from_query(NULL, buf, sizeof(buf)))
-          mutt_message(_("Failed to create query, aborting"));
+
+        // Keep copy of user's querying to name mailbox.
+        char *query_unencoded = mutt_str_strdup(buf);
+
+        if (nm_uri_from_query(NULL, buf, sizeof(buf)))
+        {
+          // Create mailbox and set name.
```
Comment (flatcap) [inline index.c]: Trailing whitespace!

**Pull request merged**